src/requiredBy.js

```diff
@@ -0,0 +1,12 @@
+export default function requiredBy(requiredByPropName, propType, defaultValue = null) {
+  return function validator(props, propName, componentName, ...rest) {
+    if (props[requiredByPropName]) {
```

Comment:
yeah in the description it says that it checks for truthiness. if it's `false`, we don't need the prop to exist.
eg. if `showFooButton` is `false`, then it doesn't need `fooButtonText` or `onFooButtonPress` to exist.